### DISCUSSION TITLE
Can I have Meson visualize the dependency graph?

### DISCUSSION THREAD

**Discussion by manuel-wagesreither** [mesonbuild/meson#10343] [Q&A] [answered]
Hi all,

I'm running CMake but am interested in migrating to Meson. CMake supports creation of a dependency graph. I find this feature very useful and would like to keep using it. Does Meson support the same? I couldn't find anything.

https://gitlab.kitware.com/cmake/community/-/wikis/doc/cmake/Graphviz
https://embeddeduse.com/2022/03/01/visualising-module-dependencies-with-cmake-and-graphviz/

Regards,
Manuel

**Comment by eli-schwartz** [comment, marked as answer]
https://ninja-build.org/manual.html#_extra_tools

Once Meson has generated the build.ninja file, you can use ninja to generate a graphviz `*.dot` file of the ninja targets, or have it spin up a tiny server that lets you browse a similar, but interactive, graph via a web browser. Does this help?

**Reply by manuel-wagesreither** [reply]
It does! Thank you a lot!

**Reply by eli-schwartz** [reply]
Cool! Sounds like another case where cmake has a built-in feature for all of its backends (make, xcode, vs, ninja) which ninja, as a much newer backend has its own built-in version of that feature. The same thing happens for compile_commands.json, which cmake creates itself, while meson tells ninja to create it.

This is primarily useful to cmake, I believe, because it implemented these before ninja existed, and there's no reason for them to remove it now.

But meson generally just says "use the ninja backend and let that create it for you". (Also meson doesn't have a make backend anyway.)

**Reply by manuel-wagesreither** [reply]
Is there a way to restrict ninja to build targets? Using the suggested command `ninja -t graph mytarget | dot -Tpng -ograph.png`, the graph includes every source file and is _huge_.
